### ISSUE TITLE
ZipArchive disposes passed stream twice

### ISSUE THREAD

**Issue by bollhals** [dotnet/runtime#79695]
### Description

I've tracked down a bug in our software where we had a stream which was not safe to call dispose multiple times (Fixed in the meantime =)). 
This surfaced when we passed said stream into a ZipArchive (reading only), because it is disposing the passed stream twice. 
I know that this is technically not a bug, but nonetheless something which can be improved.

The issue itself is caused by [this line](https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs#L167). The BinaryReader gets passed the stream, but the leaveOpen parameter is not set. Hence when [disposing](https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs#L464-L466) it disposes the stream twice.

### Reproduction Steps

- Create a erronous Stream implementation that has an issue when disposing twice.
- Create a ZipArchive in reading mode and dispose it

### Expected behavior

While not technically breaking, it still would be nice not to dispose things multiple times if not needed.

### Actual behavior

Disposes the input stream twice, once directly and once due to the BinaryReader.

### Regression?

not that I'm aware of

### Known Workarounds

Fix the wrongfully implemented stream :) 

### Configuration

I've tested it on .NET6, but it still exists on current main

**Comment by Joe4evr** [comment]
> ### Expected behavior
> While not technically breaking, it still would be nice not to dispose things multiple times if not needed.

I don't believe this needs to be the case. For years and years, [Micosoft's official guidance has been](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose):
> To help ensure that resources are always cleaned up appropriately, a `Dispose` method should be idempotent, such that it is callable multiple times without throwing an exception. Furthermore, subsequent invocations of `Dispose` should do nothing.

There's probably several places in the BCL where `Dispose` could get called multiple times (perhaps no choice in the matter), so it's fair to make the implementer responsibile following the guidelines and not throw exceptions from `Dispose`.

**Comment by bollhals** [comment]
Fully agree it doesn't need to be, but I think not doing something that is not needed is still better, even if not required. This is why I opened this issue.
